src/rest-server/src/models/v2/job/k8s.js

```diff
@@ -615,6 +649,102 @@ const getSshInfo = async (frameworkName) => {
   throw createError('Not Found', 'NoJobSshInfoError', `SSH info of job ${frameworkName} is not found.`);
 };
 
+const extractDiagnosticsInfo = (diag) => {
+  if (_.isEmpty(diag)) {
+    return null;
+  }
+  // The pattern for diagnostics is "podPattern unmatched: json" or "podPattern nmatched: json"
+  const regex = /matched: (.*)/;
+  const matches = diag.match(regex);
+  if (matches.length < 2) {
+    return null;
+  }
+  try {
+    return JSON.parse(matches[1]);
+  } catch (error) {
+    logger.warn('Get diagnostics info failed', error);
+  }
+  return null;
+};
+
+const extractRuntimeOutput = (diag) => {
+  if (_.isEmpty(diag)) {
+    return null;
+  }
+
+  // Do not container user container info
+  if (_.isNil(diag.containers) || diag.containers.length < 2) {
```

Comment:
Runtime code is in both init and app container, so why only app container has PAI_RUNTIME_ERROR_START/END?

Reply:
Actually, for init container, I just want to return specific code for specific error and do not use pattern match. Since it all controlled by ourselves. But it's OK to leverage the same exit message format with user container. Will add this part.

Reply:
ok, but it equivalent to empty exit message.

---
In reply to: [321558507](https://github.com/microsoft/pai/pull/3575#discussion_r321558507) [](ancestors = 321558507)